File: frontend/src/types/anomaly.ts
export interface MetricCollectionResponse {
  resources_scanned: number;
  datapoints_fetched: number;
  datapoints_saved: number;
}

export interface DatasetSummary {
  resource_count: number;
  total_datapoints: number;

  earliest_timestamp: string | null;
  latest_timestamp: string | null;

  minimum_points_per_resource: number;
  recommended_minimum_points: number;
  ready_for_ml: boolean;
}

export interface AnomalyDetectionResponse {
  resources_found: number;
  resources_trained: number;
  resources_skipped: number;
  points_analyzed: number;
  anomalies_detected: number;
  model_name: string;
}

export interface MetricAnomaly {
  id: number;

  resource_id: string;
  resource_name: string | null;

  metric_name: string;
  metric_timestamp: string;

  average_value: number;
  maximum_value: number;

  anomaly_score: number;
  severity: string;
  reason: string;

  model_name: string;
  is_active: boolean;
  detected_at: string;
}

export interface CurrentUser {
  id: number;
  email: string;
  role: string;
}

File: backend/app/schemas/anomaly.py
from datetime import datetime

from pydantic import BaseModel


class AnomalyDetectionResponse(BaseModel):
    resources_found: int
    resources_trained: int
    resources_skipped: int
    points_analyzed: int
    anomalies_detected: int
    model_name: str


class AnomalyResponse(BaseModel):
    id: int
    resource_id: str
    resource_name: str | None
    metric_name: str
    metric_timestamp: datetime
    average_value: float
    maximum_value: float
    anomaly_score: float
    severity: str
    reason: str
    model_name: str
    is_active: bool
    detected_at: datetime